### PR TITLE
Add the (Automated) Chart Documentation

### LINKED ISSUE #12: Provide a description for each value that can be set
### Description

Would be good to document all the values that can be set during the Helm Chart installation/update to customize it based on the user needs.

### Solution you'd like

I think we can do it in several ways, and they are all good options at the end. Those are the ones I was thinking about:

- Listing them in the [`README.md`](https://github.com/fonzdm/servarr/blob/main/README.md#values), but this would lead to a very long README and I don't like it so much;
- Creating a new markdown file maybe in `/docs` (we should then add this path to the [`.helmignore`](https://github.com/fonzdm/servarr/blob/main/.helmignore)) which will contains all the values with a short description, default value and an example.

Please, feel free to propose any other solution.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by fonzdm**
* **Please check if the PR fulfills these requirements**
- [X] The branch naming convention follows our guidelines
- [X] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

An auto-generated documentation for the chart is introduced with this PR. 
The [documentation](#./servarr/README.md) is produced based on comments of the values.yaml file, and the overall chart structure.
A pipeline takes care of fetching the chart changes and update the documentation with a dedicated commit, authored "Servarr Bot 🤖"

* **What is the current behavior?** (You can also link to an open issue here)

No documentation exists for the chart.

* **What is the new behavior (if this is a feature change)?**

Everytime someone touches the chart, the documentation is produced/updated.
Users can follow the documentation to fill a values.yaml file for their deployments.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

It's not a breaking change but it's a core functionality.

* **Other information**:

Some other refactoring will be done to the `values.yaml`, but it is not in scope of this PR.

Closes #12

### CODE REVIEW

**Comment by fonzdm** [comment]
Let me fix that linting pipeline.

File: servarr/README.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+| Key | Type | Default | Description |
+|-----|------|---------|-------------|
+| metrics.enabled | bool | `false` | Anchor to set wether to deploy the export sidecar pods or not. Requires the Prometheus stack.  Do not remove the `&metricsEnabled` anchor! |
+| volumes.downloads.name | string | `"downloads-volume"` | Name of the download pvc.  Do not remove the `&downloads-volume` anchor! |
+| volumes.downloads.size | string | `"100Gi"` | Size of the downloads volume, in Kubernets format |
+| volumes.media.name | string | `"media-volume"` | Name of the media pvc.  Do not remove the `&media-volume` anchor! |
+| volumes.media.size | string | `"250Gi"` | Size of the media volume, in Kubernets format |
+| volumes.torrentConfig.name | string | `"torrent-config"` | Name of the torrent configuration pvc.  Do not remove the `&torrentConfig` anchor! |
+| volumes.torrentConfig.size | string | `"250Mi"` | Size of the torrent configuration volume, in Kubernets format |
+
```
Comment (imgios) [inline servarr/README.md:90]: What if we move optional and required values in the other categories? Then we can add in the description if it's an optional value or not.

As an example: when reading about the storage parameters is easily to get lost

https://github.com/fonzdm/servarr/blob/58cdaeff0b8aa07f3d9d80c186a821828914b32a/servarr/README.md?plain=1#L106-L113

For the default value we have `"See the sub fields" but we are at the end of the page, and the subfields must be found first 😄

File: .github/workflows/pre-commit-hooks.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+name: 'Helm Chart Doc Generator'
+
+on:
+  push:
+    branches:
+      - 'dev'
+      - 'docs/values-description'
```
Comment (imgios) [inline .github/workflows/pre-commit-hooks.yaml:7]: Can you please replace `docs/values-description` with `main`?

File: .pre-commit-config.yaml
Comment (imgios) [inline .pre-commit-config.yaml:1]: Is this file really needed? Are you going to introduce it also as pre-commit?

File: README.md [hunk 3]
```diff
@@ -45,26 +45,23 @@ $ helm install <release-name> <servarr-chart> \
 
 ### Values
 
-| **Parameter** | **Optional** | **Description** | **Default Value** | **Example Value** |
-|---------------|--------------|-----------------|-------------------|-------------------|
-| `global.apikey` | Required | Define a common API key that will be used to automate the entire integration. | `None` | `6d9df216-cac6-4f4a-9b25-5462026ce7be` |
-| `global.storageClassName` | Required | Cluster storage class to be used for the volume creation. | `None` | `longhorn` |
-
-<!-- Table init with some values just to give the idea on how it will be. A lot of values (almost all of them) are missing and MUST be filled in the table. -->
+Please read [README.md](./servarr/README.md) for details on how to configure the values needed for this chart.
```
Comment (imgios) [inline README.md:48]: What if we change it to:

> Please, read [Chart README.md](./servarr/README.md) for details on how to configure the values needed for this chart.

the aim is to avoid any confusion with the repository readme (that they are also reading 😂)

File: .github/workflows/pre-commit-hooks.yaml
Comment (imgios) [inline .github/workflows/pre-commit-hooks.yaml:1]: Should we change the file name to anything else since it's not a pre-commit hook?

**Review by imgios — COMMENTED** [review]
Hey @fonzdm, just finished to look at it and I leaved some questions all around. Please, let me know your POV.

File: .github/workflows/pre-commit-hooks.yaml
Comment (fonzdm) [reply .github/workflows/pre-commit-hooks.yaml:1]: Right! gonna change to `helm-docs.yaml`, ok?

File: README.md (hunk 3 above)
Comment (fonzdm) [reply README.md:48]: done!

File: .github/workflows/pre-commit-hooks.yaml
Comment (imgios) [reply .github/workflows/pre-commit-hooks.yaml:1]: Sounds good!

File: servarr/README.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+| https://charts.truecharts.org | radarr | 21.2.1 |
+| https://charts.truecharts.org | sonarr | 21.2.1 |
+
+---
+
+> [!IMPORTANT]  
+> Please consider that this chart is a collection of several public helm charts.
+> These are included as sub-charts of the Servarr chart and, due to some Helm limitation, some configuration are only possible via values file.
+> For this reason, the servarr default [values.yaml](#./values.yaml) included in the chart is quite huge and it used to model the configuration of the subcharts.
+> But don't you worry! I provided some handy values, using [yaml anchros](https://medium.com/@kinghuang/docker-compose-anchors-aliases-extensions-a1e4105d70bd), to defined top-level fields.
```
[the remark this answers is not in the document]
Comment (fonzdm) [reply servarr/README.md]: Done! Changed in the gotmpl file that it used to generate the readme.

File: .github/workflows/pre-commit-hooks.yaml
Comment (fonzdm) [reply .github/workflows/pre-commit-hooks.yaml:1]: Done!

File: servarr/README.md (hunk 1 above)
Comment (fonzdm) [reply servarr/README.md:90]: Yes, I will remove Required and Optional values... I completely forgot them 😡

Comment (fonzdm) [reply servarr/README.md:84]: Done!

**Comment by fonzdm** [comment]
@imgios please check, everything resolved.

**Comment by imgios** [comment]
> @imgios please check, everything resolved.

Thank you for taking care of it. Everything seems to be fine, going to merge it into `dev` 🚀
